main.py

```diff
@@ -2,9 +2,53 @@
 import os
 import hashlib
 import sys
-load_env()
+from reportlab.pdfgen import canvas
 
 
+load_env(env_path=r'.\example.env')
+
+def is_text_file(file_path):
+    text_file_extensions = ['.txt', '.md', '.go', '.py', '.java', '.html', '.css', '.js', '.mod', '.sum']  # Add more as needed
+    return any(file_path.lower().endswith(ext) for ext in text_file_extensions)
+
+def combine_txt_files_and_create_pdf(source_directory, output_file, pdf_output, separator='**'):
+    separator_line = separator * 40 + '\n'
+    
+    # Initialize a list to store combined text
+    combined_text = []
+
+    with open(output_file, 'w', encoding='utf-8') as outfile:
+        for root, dirs, files in os.walk(source_directory):
+            for filename in files:
+                if filename.endswith('.txt'):
```

Comment:
You've defined ```is_text_file()``` method above, which accepts various file extensions, but you've hardcoded ```.txt``` here. Please make this dynamic - either by introducing a similar method that checks the extension or using the ```in``` built-in and a list

Example with ```in``` built-in:

```python
filename = os.path.basename(filename)  # example.txt
file_extension = filename.split('.')[1]  # .txt, .md, etc
```